testutil/accumulator.go

```diff
@@ -161,6 +161,17 @@ func (a *Accumulator) Get(measurement string) (*Metric, bool) {
 	return nil, false
 }
 
+// Calls the given Gather function and returns the first error found.
+func (a *Accumulator) GatherError(gf func(telegraf.Accumulator) error) error {
+	if err := gf(a); err != nil {
+		return err
+	}
+	if len(a.Errors) > 0 {
+		return a.Errors[0]
+	}
+	return nil
+}
```

Comment:
I think I would prefer to accept a [`telegraf.Input`](https://godoc.org/github.com/influxdata/telegraf#Input) interface, and then call the `Gather()` method on that interface instead of accepting the `Gather` function signature.